packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,301 @@
+   *
+   * In Bitcoin Core, transactions are sorted in the mempool using a modified
+   * feerate that considers ancestors. During block creation, transactions are
+   * considered in this order. As transactions are selected, a separate map tracks
+   * the new feerate-with-ancestors for in-mempool descendants.
+   *
+   * Our simplified approach:
+   * 1. We start with the parent transaction's fee and vsize.
+   * 2. We iteratively add inputs to the child transaction.
+   * 3. After each input addition, we recalculate the combined fee rate of the
+   *    parent and child transactions.
+   * 4. We continue adding inputs until the combined fee rate meets or exceeds
+   *    the target fee rate.
+   *
+   * This method ensures that the child transaction provides enough fee to incentivize
+   * miners to include both the parent and child transactions in a block, effectively
+   * "bumping" the fee of the parent transaction.
+   *
+   * Note: While this approach is simpler than Bitcoin Core's full implementation,
+   * it achieves the core goal of CPFP: ensuring the combined package (parent + child)
+   * is attractive for miners to include in a block.
+   */
+
+  const parentFee = new BigNumber(analysis.fee);
+  const parentVsize = analysis.vsize;
+
+  for (const utxo of availableInputs) {
+    if (
+      isCPFPFeeSatisfied(
+        parentFee,
+        parentVsize,
+        new BigNumber(childTxTemplate.currentFee),
+        childTxTemplate.estimatedVsize,
+        parseFloat(txAnalyzer.cpfpFeeRate),
+      )
+    ) {
+      break; // Stop adding inputs once CPFP fee requirements are met
+    }
+    // Skip if this UTXO is already added
+    if (
+      childTxTemplate.inputs.some(
+        (input) => input.txid === utxo.txid && input.vout === utxo.vout,
+      )
+    ) {
+      continue;
+    }
+    childTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(utxo));
+  }
+
+  // Step 7: Calculate and set the change output amount
+  const totalInputAmount = new BigNumber(childTxTemplate.getTotalInputAmount());
+  const fee = new BigNumber(childTxTemplate.targetFeesToPay);
+  const changeAmount = totalInputAmount.minus(fee);
+
+  if (changeAmount.lte(dustThreshold)) {
+    if (strict) {
+      throw new Error(
+        "Change amount is below the dust threshold. Increase inputs or reduce fee rate.",
+      );
+    } else {
+      console.warn(
+        "Change amount is below dust threshold. Adjusting transaction.",
+      );
+
+      childTxTemplate.adjustChangeOutput();
+    }
+  } else {
+    childTxTemplate.outputs[0].setAmount(changeAmount.toString());
+  }
+
+  childTxTemplate.outputs[0].setAmount(changeAmount.toString());
+
+  // Step 8: Validate the child transaction
+  if (!childTxTemplate.validate()) {
+    throw new Error(
+      "Failed to create a valid CPFP transaction. Ensure all inputs and outputs are valid and fee requirements are met.",
+    );
+  }
+
+  // Step 9: Ensure the combined (parent + child) fee rate meets the target
```

Comment:
And this could maybe be `_validateCPFPPackage(txAnalyzer, childTxTemplate)`

Reply:
[added](https://github.com/caravan-bitcoin/caravan/pull/114/commits/ed9f6c70ed681c17d1ed07b444329b98811948db#diff-852fd83c241ed78594dc29389ec04d1425d784e665c706434a8de046f7daa85dR287)